File: TerminBA/TerminBA.Models/Request/ReservationInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TerminBA.Models.Request
{
    public class ReservationInsertRequest
    {
        public int? UserId { get; set; }

        public int? FacilityId { get; set; }

        [Required]
        public DateOnly ReservationDate { get; set; }

        [Required]
        public TimeOnly StartTime { get; set; }

        [Required]
        public TimeOnly EndTime { get; set; }

        [Required]
        [MaxLength(100)]
        public string? Status { get; set; }

        public int? ChosenSportId { get; set; }
    }
}

File: TerminBA/TerminBA.UI/terminba_mobile/lib/model/reservation_insert_request.dart
/// Matches the backend `ReservationInsertRequest`.
///
/// `reservationDate` → `DateOnly` on server (send as `"YYYY-MM-DD"`).
/// `startTime` / `endTime` → `TimeOnly` on server (send as `"HH:MM:SS"`).
class ReservationInsertRequest {
  final int? userId;
  final int? facilityId;
  final String reservationDate; // "YYYY-MM-DD"
  final String startTime;       // "HH:MM:SS"
  final String endTime;         // "HH:MM:SS"
  final double price;
  final int? chosenSportId;
  final String? paymentMethod;

  const ReservationInsertRequest({
    this.userId,
    this.facilityId,
    required this.reservationDate,
    required this.startTime,
    required this.endTime,
    required this.price,
    this.chosenSportId,
    this.paymentMethod,
  });

  Map<String, dynamic> toJson() => {
        if (userId != null) 'userId': userId,
        if (facilityId != null) 'facilityId': facilityId,
        'reservationDate': reservationDate,
        'startTime': startTime,
        'endTime': endTime,
        'price': price,
        if (chosenSportId != null) 'chosenSportId': chosenSportId,
        if (paymentMethod != null) 'paymentMethod': paymentMethod,
      };
}
